apps/condo/domains/meter/constants/registerMetersReadingsService.js

```diff
@@ -0,0 +1,61 @@
+const DATE_SYMBOLS_ORDER_VARIANTS = [
+    ['YYYY', 'MM', 'DD'],
+    ['DD', 'MM', 'YYYY'],
+    ['MM', 'YYYY'],
+    ['YYYY', 'MM'],
+]
+const DATE_SYMBOLS_DELIMITERS = ['/', '.', '-']
+const TIME_VARIANTS = [
+    'HH:mm:ss',
+    'HH:mm',
+]
+
+const DATE_FORMATS = DATE_SYMBOLS_ORDER_VARIANTS
+    .flatMap((order) => DATE_SYMBOLS_DELIMITERS.map(delimiter => order.join(delimiter)))
+const DATE_TIME_FORMATS = DATE_FORMATS
+    .flatMap((dateFormat) => TIME_VARIANTS.map(timeVariant => [dateFormat, timeVariant].join(' ')))
+
+const ISO_DATE_FORMAT = 'YYYY-MM-DD'
+const EUROPEAN_DATE_FORMAT = 'DD.MM.YYYY'
+const DATE_PARSING_FORMATS = [
+    ...DATE_TIME_FORMATS,
+    ...DATE_FORMATS,
+
+    'YYYY-MM-DDTHH:mm:ss.SSS[Z]', // The result of dayjs().toISOString()
+    'YYYY-MM-DDTHH:mm:ss.SSSZ',
+    'YYYY-MM-DDTHH:mm:ss.SSS',
+    'YYYY-MM-DDTHH:mm:ssZZ',
+    'YYYY-MM-DDTHH:mm:ssZ',
+    'YYYY-MM-DDTHH:mm:ss',
+].sort((a, b) => b.length - a.length) // Order matters! see "Differences to moment" https://day.js.org/docs/en/parse/string-format
+
+const READINGS_LIMIT = 500
+
+const DATE_FIELD_PATHS = [
```

Comment:
Maybe use date fields from schema file instead?

Reply:
I assume you mean that in MeterReading and Meter schema we setting "isRequired" to fields and could use them. But there is a cool circular dependency happening through meter/tasks. So maybe keep it as is

Reply:
Does this mean, that after adding a new field to meterReading, I should also add it in paths, and in translations...

I guess it's not critical here and can be skipped, since it's not frequently-changed model, but it seems strange for me. I've already got required fields and names in schema. why do I need to redefine new ones?

Reply:
If you add date field in schema it does not mean that it automatically starts parsing in registerMetersReadings or import, does not mean that translations in /lang appeared by themselves. So you will check custom mutation and task to modify logic, see that translations / validations depending on these variables and will modify them.

So i don't thinks there is any issue, maybe this even will make you change only 2 lines here, instead of 4 lines in different files